File: seed/csharp-model/trace/src/SeedTrace/V2/Problem/FunctionSignature.cs
using SeedTrace.V2;
using System.Text.Json.Serialization;

namespace SeedTrace.V2;

public class FunctionSignature
{
    public class _Void : VoidFunctionSignature
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "void";
    }
    public class _NonVoid : NonVoidFunctionSignature
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "nonVoid";
    }
    public class _VoidThatTakesActualResult : VoidFunctionSignatureThatTakesActualResult
    {
        [JsonPropertyName("type")]
        public string Type { get; } = "voidThatTakesActualResult";
    }
}

File: seed/ruby-sdk-v2/trace/lib/seed/v_2/v_3/problem/function_signature.rb
# frozen_string_literal: true

module Seed
    module Types
        class FunctionSignature < Internal::Types::Union

            discriminant :type

            member -> { Seed::V2::V3::Problem::VoidFunctionSignature }, key: "VOID"
            member -> { Seed::V2::V3::Problem::NonVoidFunctionSignature }, key: "NON_VOID"
            member -> { Seed::V2::V3::Problem::VoidFunctionSignatureThatTakesActualResult }, key: "VOID_THAT_TAKES_ACTUAL_RESULT"
    end
end
